File: Android/java/doraemonkit/src/main/java/com/didichuxing/doraemonkit/util/LogHelper.java
package com.didichuxing.doraemonkit.util;

import com.didichuxing.doraemonkit.util.LogUtils;

/**
 * Created by wanglikun on 2018/9/10.
 */

public class LogHelper {
    private static final String TAG = "DoraemonKit";
    private static boolean IS_DEBUG = true;

    public static void d(String subTag, String msg) {
        if (!IS_DEBUG) {
            return;
        }
        LogUtils.v("[" + subTag + "]: " + msg);
    }

    public static void i(String subTag, String msg) {
        if (!IS_DEBUG) {
            return;
        }
        LogUtils.v("[" + subTag + "]: " + msg);
    }

    public static void e(String subTag, String msg) {
        if (!IS_DEBUG) {
            return;
        }
        LogUtils.v("[" + subTag + "]: " + msg);
    }

    public static void json(String subTag, Object o) {
        if (!IS_DEBUG) {
            return;
        }
        LogUtils.json("[" + subTag + "]: ", o);
    }

    public static void setDebug(boolean debug) {
        IS_DEBUG = debug;
    }
}

File: Android/kotlin/doraemonkit/src/main/java/com/didichuxing/doraemonkit/util/LogHelper.kt
package com.didichuxing.doraemonkit.util

import com.blankj.utilcode.util.LogUtils

/**
 * Created by wanglikun on 2018/9/10.
 */
object LogHelper {
    private const val TAG = "DoraemonKit"
    private var IS_DEBUG = true
    fun d(subTag: String, msg: String) {
        if (!IS_DEBUG) {
            return
        }
        LogUtils.v("[$subTag]: $msg")
    }

    fun i(subTag: String, msg: String) {
        if (!IS_DEBUG) {
            return
        }
        LogUtils.v("[$subTag]: $msg")
    }

    fun e(subTag: String, msg: String) {
        if (!IS_DEBUG) {
            return
        }
        LogUtils.v("[$subTag]: $msg")
    }

    fun setDebug(debug: Boolean) {
        IS_DEBUG = debug
    }
}